# 6 - Comparação de modelos

Primeiramente vamos rever a lista de modelos existentes no **Google Gemini**.

In [ ]:
for m in modelos:
    if 'generateContent' in m.supported_generation_methods:
        print(m.name)

models/gemini-1.0-pro
models/gemini-1.0-pro-001
models/gemini-1.0-pro-latest
models/gemini-1.0-pro-vision-latest
models/gemini-1.5-flash
models/gemini-1.5-flash-001
models/gemini-1.5-flash-latest
models/gemini-1.5-pro
models/gemini-1.5-pro-001
models/gemini-1.5-pro-latest
models/gemini-pro
models/gemini-pro-vision


### Modelo Gemini 1.5 Flash

O Gemini 1.5 Flash é um modelo multimodal rápido e versátil, projetado para escalonamento em diversas tarefas. Este modelo pode lidar com entradas em diferentes formatos, oferecendo saídas textuais eficientes.

#### Detalhes do Modelo

#### Propriedade e Descrição

- **Código do modelo**: `gemini-1.5-flash-latest`
- **Entradas**: Áudio, imagens, vídeo e texto
- **Saída**: Texto
- **Métodos de geração com suporte**: `generateContent`
- **Versão mais recente**: `gemini-1.5-flash-latest`
- **Versão estável mais recente**: `gemini-1.5-flash`

#### Limites e Capacidades

- **Limite de tokens de entrada**: 1.048.576
- **Limite de tokens de saída**: 8.192
- **Número máximo de imagens por comando**: 3.600
- **Duração máxima do vídeo**: 1 hora
- **Duração máxima do áudio**: Aproximadamente 9,5 horas
- **Número máximo de arquivos de áudio por comando**: 1

#### Limites de Taxa

- **Sem custo financeiro**:
  - **15 RPM**: Requisições por minuto
  - **1 milhão TPM**: Tokens por mês
  - **1.500 RPD**: Requisições por dia

- **Pay-as-you-go**:
  - **360 RPM**: Requisições por minuto
  - **10 milhões TPM**: Tokens por mês
  - **10.000 RPD**: Requisições por dia

#### Funcionalidades Adicionais

- **Instruções do sistema**: Compatível
- **Modo JSON**: Compatível


### Modelo Gemini 1.5 Pro

O Gemini 1.5 Pro é um modelo multimodal avançado, projetado para lidar com diversas entradas e fornecer saídas textuais eficientes. Este modelo oferece capacidades robustas para processamento de áudio, imagens, vídeo e texto.

#### Detalhes do Modelo

#### Propriedade e Descrição

- **Código do modelo**: `models/gemini-1.5-pro-latest`
- **Entradas**: Áudio, imagens, vídeo e texto
- **Saída**: Texto
- **Métodos de geração com suporte**: `generateContent`
- **Versão mais recente**: `gemini-1.5-pro-latest`
- **Versão estável mais recente**: `gemini-1.5-pro`
- **Atualização mais recente**: Abril de 2024

#### Limites e Capacidades

- **Limite de tokens de entrada**: 1.048.576
- **Limite de tokens de saída**: 8.192
- **Número máximo de imagens por comando**: 3.600
- **Duração máxima do vídeo**: 1 hora
- **Duração máxima do áudio**: Aproximadamente 9,5 horas
- **Número máximo de arquivos de áudio por comando**: 1

#### Limites de Taxa

- **Sem custo financeiro**:
  - **2 RPM**: Requisições por minuto
  - **32.000 TPM**: Tokens por mês
  - **50 RPD**: Requisições por dia
  - **46.080.000 TPD**: Tokens por dia

- **Pay-as-you-go**:
  - **360 RPM**: Requisições por minuto
  - **10 milhões TPM**: Tokens por mês
  - **10.000 RPD**: Requisições por dia
  - **14.400.000.000 TPD**: Tokens por dia

  - **Contexto de dois milhões**:
  - **1 RPM**: Requisições por minuto
  - **2 milhões TPM**: Tokens por mês
  - **50 RPD**: Requisições por dia

#### Funcionalidades Adicionais

- **Instruções do sistema**: Compatível
- **Modo JSON**: Compatível


Para mais informações acesse o seguinte site: [Google Gemini models.](https://ai.google.dev/gemini-api/docs/models/gemini?hl=pt-br)

**Restante Inserir a área de comparação com ChatGPT**

**Caso seja aprovado criar o requirements.txt**

Como visto vou pegar modelos mais atualizados (No momento dessa publicação) os 1.5 em sua verção Pro e Flash.